### PR TITLE
fix: Migrate API usage from *.metafi and *.metaswap subdomains to *.api.cx.metamask.io

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## **Description**

API Platform team intends to deprecate *.metafi and *.metaswap subdomains in favour of *.api.cx.metamask.io
The new *.api.cx.metamask.io is already in place.
The old *.metafi and *.metaswap are still used across the controllers.
Controllers do not break currently because redirects are in place for the older sub-domains.
The redirects will be removed once all clients (Extension, Mobile, Core) are migrated.

## **Related issues**

Fixes: [#2351](https://github.com/MetaMask/MetaMask-planning/issues/2351)

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by georgewrmarshall — COMMENTED** [review]
The updates to the component-library folder look great! There's one failing unit test and some possibly flaky e2e tests to address. I can approve on behalf of @MetaMask/design-system-engineers once those are resolved.

- Verified that new image links in `AvatarToken` are working. ✅

File: package.json [hunk 1]
```diff
@@ -282,7 +282,7 @@
     "@metamask/address-book-controller": "^4.0.1",
     "@metamask/announcement-controller": "^6.1.0",
     "@metamask/approval-controller": "^6.0.1",
-    "@metamask/assets-controllers": "^29.0.0",
+    "@metamask/assets-controllers": "^30.0.0",
```
Comment (bergeron) [inline package.json:285]: This will require other changes, as you found form https://github.com/MetaMask/metamask-extension/pull/24913

Comment (cryptodev-2s) [reply package.json:285]: yes indeed this PR is blocked by the assets-controllers bump being currently worked on #24913

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

**Comment by cryptodev-2s** [comment]
> The updates to the component-library folder look great! There's one failing unit test and some possibly flaky e2e tests to address. I can approve on behalf of @MetaMask/design-system-engineers once those are resolved.
> 
> * Verified that new image links in `AvatarToken` are working. ✅

Pipeline is back to green

**Comment by cryptodev-2s** [comment]
@SocketSecurity ignore npm/@metamask/name-controller@6.0.1

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🚀 Approving on behalf of @MetaMask/design-system-engineers 
- Checked to ensure no remaining instances of `metafi.codefi.network` in codebase ✅ 
- Checked for console errors ✅ 
- Check some image links in storybook and app render as expected ✅ 

Screenshot 2024-06-05 at 1 17 17 PM

**Review by kanthesha — APPROVED** [review]
LGTM.

**Review by mikesposito — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by Prithpal-Sooriya — APPROVED** [review]
Good on the notifications side (we are not using any of the old API domains)

**Review by pedronfigueiredo — APPROVED** [review]
I reviewed the privacy snapshots. LGTM!

**Pull request merged**